astor/code_gen.py

```diff
@@ -81,34 +112,108 @@ def set_precedence(value, *nodes):
 set_precedence = precedence_setter()
 
 
+def conditional(*params):
+    """ If the last param evaluates to None, just return
+        an empty tuple
+    """
+    return params if params[-1] is not None else ()
+
+
+def comma_group(items, trailing=False,
+                # Constants
+                set_precedence=set_precedence,
+                Comma=Precedence.Comma, enumerate=enumerate):
+    """ Print a group of 0 or more items, with commas between,
+        and an optional trailing comma.
+    """
+    set_precedence(Comma, *items)
+    yield (((', ', item) if idx else item)
+                for idx, item in enumerate(items))
+    if trailing:
+        yield ','
+
+
+def else_body(node,
+              # Constants
+              If=ast.If, set_precedence=set_precedence,
```

Comment:
But yeah, overall, I'd say that the gains from doing this are much smaller.  Attribute lookups must be much faster than they used to be, because one of the things that really surprised me was that the gains from using closures are much smaller now, and non-existent or negative in a few corner, presumably for the same reason.  You have to have a lot of attribute accesses to offset the cost of building the closure.